java/arcs/android/devtools/StoreOperationMessage.kt

```diff
@@ -0,0 +1,148 @@
+package arcs.android.devtools
+
+import arcs.android.devtools.DevToolsMessage.Companion.ACTOR
+import arcs.android.devtools.DevToolsMessage.Companion.ADDED
+import arcs.android.devtools.DevToolsMessage.Companion.ADD_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.CLEAR_ALL_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.CLEAR_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.CLOCK
+import arcs.android.devtools.DevToolsMessage.Companion.FAST_FORWARD_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.OLD_CLOCK
+import arcs.android.devtools.DevToolsMessage.Companion.OPERATIONS
+import arcs.android.devtools.DevToolsMessage.Companion.REMOVED
+import arcs.android.devtools.DevToolsMessage.Companion.REMOVE_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.STORE_ID
+import arcs.android.devtools.DevToolsMessage.Companion.STORE_OP_MESSAGE
+import arcs.android.devtools.DevToolsMessage.Companion.STORE_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.UPDATE_TYPE
+import arcs.android.devtools.DevToolsMessage.Companion.VALUE
+import arcs.core.common.Referencable
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtEntity
+import arcs.core.crdt.CrdtOperation
+import arcs.core.crdt.CrdtSet
+import arcs.core.crdt.CrdtSingleton
+import arcs.core.storage.ProxyMessage
+import arcs.core.util.JsonValue
+
+/**
+ * An implementation of [StoreMessage] to inform DevTools that a [Store] received a
+ * [CrdtOperation].
+ */
+class StoreOperationMessage(
+    private val actualMessage: ProxyMessage.Operations<CrdtData, CrdtOperation, Any?>,
+    private val storeType: String
+) : StoreMessage {
+    override val kind: String = STORE_OP_MESSAGE
+    override val message: JsonValue<*>
+        get() = JsonValue.JsonObject(
+            STORE_ID to JsonValue.JsonNumber(actualMessage.id?.toDouble() ?: 0.0),
+            STORE_TYPE to JsonValue.JsonString(storeType),
+            OPERATIONS to JsonValue.JsonArray(getMessageAsList())
+        )
+
+    /**
+     * Turn the ProxyMessage into a List of JsonValues.
+     */
+    private fun getMessageAsList(): List<JsonValue<*>> {
```

Comment:
Similar thoughts regarding naming semantics: maybe something like `asJsonValueList`? It's less important on a private method to get this right, so just a nit here.